pkg/autogen/v2/autogen.go

```diff
@@ -301,3 +301,46 @@ func (a *ImplAutogenV2) ExtractPodSpec(resource unstructured.Unstructured) (*uns
 
 	return &unstructured.Unstructured{Object: podSpec}, nil
 }
+
+type ExtractPodFunc func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error)
+
+func GetPodExtractor(kind string) ExtractPodFunc {
+	return func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+		autogen := &ImplAutogenV2{}
+		return autogen.ExtractPodSpec(*resource)
+	}
+}
+
+func ComputeRules(p kyvernov1.PolicyInterface, kind string) ([]kyvernov1.Rule, ExtractPodFunc) {
+	spec := p.GetSpec()
+	applyAutoGen, desiredControllers := CanAutoGen(spec)
+	if !applyAutoGen {
+		desiredControllers = sets.New("none")
+	}
+
+	var actualControllers sets.Set[string]
+	ann := p.GetAnnotations()
+	actualControllersString, ok := ann[kyverno.AnnotationAutogenControllers]
+	if !ok || !applyAutoGen {
+		actualControllers = desiredControllers
+	} else {
+		if !applyAutoGen {
+			actualControllers = desiredControllers
+		} else {
+			actualControllers = sets.New(strings.Split(actualControllersString, ",")...)
+		}
```

Comment:
You should simplify this, you dont need to create another set, you don't have to check applyAutoGen in multiple places